src/Metadata/Driver/XmlDriver.php

```diff
@@ -68,6 +69,7 @@ protected function loadMetadataFromFile(\ReflectionClass $class, string $path):
         $exclude = $elem->attributes()->exclude;
         $excludeAll = null !== $exclude ? 'true' === strtolower((string) $exclude) : false;
         $classAccessType = (string) ($elem->attributes()->{'access-type'} ?: PropertyMetadata::ACCESS_TYPE_PROPERTY);
+        $readOnlyClass = $elem->attributes()->{'read-only'};
```

Comment:
for some weird reason I really cannot find out, I cannot do parseExpression() here, because if re-use the variable $readOnlyClass in the property-loop it is only working for the first call. On the second call then, I'm getting error on unserializing the cache. But if I do parseExpression() inside the property-loop it works fine.
Makes really no sense to me.

Reply:
you should cast to string the xml node

Reply:
If you look into my previous commit, I'm doing that there.
I've noticed that it was actually only working for the caching-call. The second call fails with error:

> Return value of JMS\Serializer\Metadata\PropertyMetadata::unserializeProperties() must be of the type string, null returned [] []

It's really weird because if I do the exact same thing inside of the property-loop, it is working.
So putting this block line 90-96 (previous commit):
`        $readOnlyClass = false;
        $readOnlyIfClass = null;
        if(null === $readOnlyIf = $elem->attributes()->{'read-only-if'}) {
            $readOnlyClass = 'true' === strtolower((string) $elem->attributes()->{'read-only'});
        } else {
            $readOnlyIfClass = $this->parseExpression((string) $readOnlyIf);
        }`

into the loop starting at line 174, it works 

It's just confusing me and I wanted to find out how this happens.
But it does not really matter because doing parseExpression on every porperty will maybe only affect the performance on when creating cache